sdk/tests/core/test_az_json.c

```diff
@@ -228,6 +228,110 @@ static void test_json_writer(void** state)
+
+      az_span_to_str((char*)array, 200, az_json_writer_get_bytes_used_in_destination(&writer));
+
+      assert_string_equal(array, "{\"name\":  \"f\\u0065o\", \"values\": [1, 2, 3,{}]}");
+    }
+
+    {
+      TEST_EXPECT_SUCCESS(az_json_writer_init(&writer, AZ_SPAN_FROM_BUFFER(array), NULL));
+      TEST_EXPECT_SUCCESS(az_json_writer_append_begin_object(&writer));
+      TEST_EXPECT_SUCCESS(az_json_writer_append_property_name(&writer, AZ_SPAN_FROM_STR("name")));
+      TEST_EXPECT_SUCCESS(az_json_writer_append_bool(&writer, true));
+
+      TEST_EXPECT_SUCCESS(az_json_writer_append_property_name(&writer, AZ_SPAN_FROM_STR("foo")));
+      TEST_EXPECT_SUCCESS(az_json_writer_append_json_text(
+          &writer, AZ_SPAN_FROM_STR("[\"bar\",null,0,-12,12,9007199254740991]")));
+
+      TEST_EXPECT_SUCCESS(az_json_writer_append_end_object(&writer));
+
+      az_span_to_str((char*)array, 200, az_json_writer_get_bytes_used_in_destination(&writer));
+
+      assert_string_equal(
+          array,
+          "{"
+          "\"name\":true,"
+          "\"foo\":[\"bar\",null,0,-12,12,9007199254740991]"
+          "}");
+    }
+  }
+}
+
+static void test_json_writer_append_nested_invalid(void** state)
+{
+  (void)state;
+  {
+    uint8_t array[200] = { 0 };
+    az_json_writer writer = { 0 };
+
+    {
+      TEST_EXPECT_SUCCESS(az_json_writer_init(&writer, AZ_SPAN_FROM_BUFFER(array), NULL));
+      assert_int_equal(
+          az_json_writer_append_json_text(&writer, AZ_SPAN_FROM_STR("\"name\":  \"f\\u0065o\"")),
+          AZ_ERROR_UNEXPECTED_CHAR);
+    }
+    {
+      TEST_EXPECT_SUCCESS(az_json_writer_init(&writer, AZ_SPAN_FROM_BUFFER(array), NULL));
+      assert_int_equal(
+          az_json_writer_append_json_text(&writer, AZ_SPAN_FROM_STR("{]")),
+          AZ_ERROR_UNEXPECTED_CHAR);
+    }
+    {
+      TEST_EXPECT_SUCCESS(az_json_writer_init(&writer, AZ_SPAN_FROM_BUFFER(array), NULL));
+      TEST_EXPECT_SUCCESS(az_json_writer_append_begin_object(&writer));
+      TEST_EXPECT_SUCCESS(az_json_writer_append_property_name(&writer, AZ_SPAN_FROM_STR("name")));
+      TEST_EXPECT_SUCCESS(az_json_writer_append_bool(&writer, true));
+
+      TEST_EXPECT_SUCCESS(az_json_writer_append_property_name(&writer, AZ_SPAN_FROM_STR("foo")));
+      assert_int_equal(
+          az_json_writer_append_json_text(
+              &writer, AZ_SPAN_FROM_STR("\"bar\",null,0,-12,12,9007199254740991")),
+          AZ_ERROR_UNEXPECTED_CHAR);
+    }
+    {
+      TEST_EXPECT_SUCCESS(az_json_writer_init(&writer, AZ_SPAN_FROM_BUFFER(array), NULL));
+      assert_int_equal(
+          az_json_writer_append_json_text(&writer, AZ_SPAN_FROM_STR("{\"name")),
+          AZ_ERROR_UNEXPECTED_END);
+    }
+    /*{
+      TEST_EXPECT_SUCCESS(az_json_writer_init(&writer, AZ_SPAN_FROM_BUFFER(array), NULL));
+      assert_int_equal(
+          az_json_writer_append_json_text(&writer, AZ_SPAN_FROM_STR("{\"name\":  ")),
+          AZ_ERROR_UNEXPECTED_END);
+    }*/
```

Comment:
Waiting on https://github.com/Azure/azure-sdk-for-c/pull/1115 to fix https://github.com/Azure/azure-sdk-for-c/issues/1114 before un-commenting this test.